torch/csrc/jit/tensorexpr/half_support.h

```diff
@@ -0,0 +1,71 @@
+#include <torch/csrc/jit/tensorexpr/ir.h>
+#include <torch/csrc/jit/tensorexpr/ir_printer.h>
+#include <torch/csrc/jit/tensorexpr/ir_visitor.h>
+#include <torch/csrc/jit/tensorexpr/tensor.h>
+
+namespace torch {
+namespace jit {
+namespace tensorexpr {
+
+// Walk the Statment looking for Half size loads/stores.
+class HalfChecker : public IRMutator {
+ public:
+  bool hasHalf() {
+    return hasHalf_;
+  }
+
+  const Expr* mutate(const Load* v) override {
+    const Expr* child = IRMutator::mutate(v);
+    if (child->dtype().scalar_type() != ScalarType::Half) {
+      return child;
+    }
+
+    hasHalf_ = true;
+
+    return new Cast(Dtype(kFloat, child->dtype().lanes()), child);
+  }
+
+  Stmt* mutate(const Store* v) override {
+    const Expr* new_val = v->value()->accept_mutator(this);
+
+    if (v->value()->dtype().scalar_type() == ScalarType::Half) {
+      // TODO discards lanes.
```

Comment:
Hmmm is this `discard lanes` comment still true, or is it stale?  The line below seems to preserve lanes, but...?

Reply:
Should be fixed in master.